### PR TITLE
Update sbt to latest version.

### LINKED ISSUE #2102: Update build to sbt 1.3.x
To keep slick up to date with scala ecosystem. I have a WIP branch on this matter.

### PR DESCRIPTION
**Pull request by alexFrankfurt**
Closes #2102.
updates to 1.4.7
~~Is blocked by sbt/sbt#5459.~~(2)

This PR changes syntax to new slash style. 

Since Project#addSbtFiles was removed in sbt/sbt#3068 settings for sample projects get overriden by using Extracted#appendWithSession, until sbt/sbt#5459 gets fixed changes only remain for last project they being applied to. sampleOverridden AttributeKey needed to short-circuit infinite loop more details is in sbt/sbt#3544.

(2) Splits tests for samples to 4 different sbt tasks, before each task command is run to update corresponding sample sbt settings.

[Ornate](https://github.com/szeiger/ornate)-0.6 library added to libraryDependencies by plugin seems to only be published for 2.12 (https://mvnrepository.com/artifact/com.novocode/ornate), ~~therefore I temporarily set scalaVersion to 2.12. 
It needs to be published for 2.11 and 2.13 @szeiger?~~

Make sure:
- [x] everything compiles
- [x] tests pass. (Travis CI)
- [x] osgi 
  - [x] (slick / osgiBundle)
  - [x] (osgitests / Test / test) (2)(adds packages To Export-Package in 4ffc6ee)
- [x] docs build. (++ 2.12.12 ; doc)(2)(docs only built on 2.12 in admin/build.sh)

### CODE REVIEW

**Comment by szeiger** [comment]
Ornate should always use 2.12 for its own config but this is broken and I haven't found any way to work around it yet. I recently ran into the same issue again with benchdb. The workaround is to only run it when the version is set to 2.12. There's no point generating the same docs for multiple versions anyway.

**Comment by SethTisue** [comment]
@alexFrankfurt interested in returning to this and trying to land it?

**Comment by SethTisue** [comment]
CI results: green on 2.12 + JDK 8, the others fail

**Comment by alexFrankfurt** [comment]
@SethTisue ping

File: build.sbt [hunk 1]
```diff
@@ -91,17 +151,19 @@ lazy val sampleSlickTestkitExampleProject =
...
+    slickTestkitProject / Test / test,
+    slickTestkitProject / DocTest / test,
+    osgiTestProject / Test / test,
+    reactiveStreamsTestProject / Test / test,
+    slickProject / Compile / packageDoc,
+    slickCodegenProject / Compile / packageDoc,
+    slickHikariCPProject / Compile / packageDoc,
+    slickTestkitProject / Compile / packageDoc,
+    slickProject / Compile / mimaReportBinaryIssues // enable for minor versions
+//    aRootProject / testSamples
```
Comment (SethTisue) [inline build.sbt:166]: should we be concerned that this is now commented out?

File: project/build.properties [hunk 2]
```diff
@@ -1,2 +1,2 @@
 # Remember to change the sbt version in all sample projects when updating it here
-sbt.version=0.13.18
+sbt.version=1.4.5
```
Comment (SethTisue) [inline project/build.properties:2]: is there a reason not to go all the way to 1.4.7?

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,8 +1,14 @@
 scalacOptions ++= Seq("-deprecation", "-feature")
 
-addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.3.0")
-addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.7.0")
-addSbtPlugin("com.typesafe" % "sbt-sdlc" % "0.1")
-addSbtPlugin("de.johoop" % "sbt-testng-plugin" % "3.1.1")
-addSbtPlugin("com.novocode" % "sbt-ornate" % "0.5")
 addSbtPlugin("com.jsuereth" % "sbt-pgp" % "2.0.2")
+addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.7.0")
```
Comment (SethTisue) [inline project/plugins.sbt:4]: a newer MiMa is available -- I guess that could be a separate PR

File: samples/hello-slick/project/build.properties [hunk 4]
```diff
@@ -1 +1 @@
-sbt.version=0.13.18
+sbt.version=1.3.8
```
Comment (SethTisue) [inline samples/hello-slick/project/build.properties:1]: above you went to 1.4.5, is it just an oversight this isn't the same?

Comment (SethTisue) [reply samples/hello-slick/project/build.properties:1]: ditto on the other samples

Comment (alexFrankfurt) [reply samples/hello-slick/project/build.properties]: oversight

Comment (alexFrankfurt) [reply samples/hello-slick/project/build.properties]: changed to 1.4.7

File: project/plugins.sbt (hunk 3 above)
Comment (alexFrankfurt) [reply project/plugins.sbt]: I've set the latest version and for sbt-pgp

File: project/build.properties (hunk 2 above)
Comment (alexFrankfurt) [reply project/build.properties]: no, I've updated

File: build.sbt (hunk 1 above)
Comment (alexFrankfurt) [reply build.sbt:166]: `testSample1,2,3,4` contain the same tests as `testSamples`, `updateSampleHelloSlick` updates settings for `sample-hello-slick` (scalaVersion and others) and testSample1 tests it, same for other 3 samples, these arguments passed to sbt for Travis CI [here: build.sh](https://github.com/alexFrankfurt/slick/blob/update-sbt/admin/build.sh#L34). And to test locally: `updateSampleHelloSlick; root/testSample1`

Comment (SethTisue) [reply build.sbt:166]: okay, I don't really follow because I don't know the code, but if you're satisfied, I'm satisfied :-)

**Review by SethTisue — APPROVED** [review]
LGTM. Let's see if any other reviewers have anything to say about it. If they don't, ping me at the end of the week and I'll merge it.

File: appveyor.yml [hunk 5]
```diff
@@ -4,7 +4,7 @@ install:
       Add-Type -AssemblyName System.IO.Compression.FileSystem
       if (!(Test-Path -Path "C:\sbt" )) {
         (new-object System.Net.WebClient).DownloadFile(
-          'https://piccolo.link/sbt-0.13.18.zip',
+          'https://piccolo.link/sbt-1.3.8.zip',
```
Comment (pjfanning) [inline appveyor.yml:7]: should this be 1.4.7 to match the build.properties?

Comment (alexFrankfurt) [reply appveyor.yml:7]: agree, probably switch to https://github.com/sbt/sbt/releases/download/v1.4.7/sbt-1.4.7.zip following the example https://github.com/sbt/sbt/issues/5860#issuecomment-698335998 since piccolo returns 404 for 1.4.x

Comment (alexFrankfurt) [reply appveyor.yml]: switched

**Comment by SethTisue** [comment]
🎉

**Comment by alexFrankfurt** [comment]
🎉

**Comment by SethTisue** [comment]
@alexFrankfurt `osgitests/test` currently fails with 

```
[info] Test slick.osgi.test.BasicTest.testPlainSQL started
[error] Test slick.osgi.test.BasicTest.testPlainSQL failed: java.lang.IllegalArgumentException: invalid range "${range;[==,=+)}": invalid format, took 0.182 sec
[error]     at org.osgi.framework.VersionRange.this(VersionRange.java:146)
[error]     at org.apache.felix.framework.util.manifestparser.ManifestParser.normalizeImportClauses(ManifestParser.java:329)
[error]     at org.apache.felix.framework.util.manifestparser.ManifestParser.this(ManifestParser.java:183)
```

any idea what that's about...?

(This came up over at #2175, where I'm working on re-enabling CI in this repo.)

**Comment by alexFrankfurt** [comment]
@SethTisue 
```
Import-Package: scala;version="[2.12,2.13)",scala.annotation;version="
 [2.12,2.13)",scala.collection;version="[2.12,2.13)",scala.collection.
 compat;version="${range;[==,=+)}",scala.collection.convert;version="[
```
Version for scala-collection-compat couldn't be rendered, since from 2.1.6 bundle was dropped: scala/scala-collection-compat#321, it's possible to omit scala-collection-compat from Import-Package then osgitests/test fails:
```
[info] Test run started
*** (o.o.p.e.junit.impl.ProbeRunner) running test class slick.osgi.test.BasicTest
[info] Test slick.osgi.test.BasicTest.testPlainSQL started
Exception in thread "AsyncExecutor.default-1" java.lang.NoClassDefFoundError: scala/collection/compat/package$
  | => oat slick.jdbc.StatementInvoker.results(StatementInvoker.scala:45)
        at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:22)
        at slick.jdbc.Invoker.first(Invoker.scala:30)
        at slick.jdbc.Invoker.first$(Invoker.scala:29)
        at slick.jdbc.StatementInvoker.first(StatementInvoker.scala:16)
        at slick.jdbc.StreamingInvokerAction$HeadAction.run(StreamingInvokerAction.scala:52)
        at slick.jdbc.StreamingInvokerAction$HeadAction.run(StreamingInvokerAction.scala:51)
        at slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276)
        at slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: scala.collection.compat.package$ not found by com.typesafe.slick [13]
        at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1597)
        at org.apache.felix.framework.BundleWiringImpl.access$300(BundleWiringImpl.java:79)
        at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:1982)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 12 more

```
when added to Export-Package https://github.com/alexFrankfurt/slick/commit/79010b2cfadae57d203499234e2e81902b5f5596, tests succeed, but scala/collection gets added to publishLocal artifacts.

**Comment by SethTisue** [comment]
@alexFrankfurt thank you for investigating! I've opened a ticket proposing we drop OSGi support: https://github.com/slick/slick/issues/2176 — let's discuss further there
